### PR TITLE
drivers/ieee802154 Kconfig and build cleanup

### PR DESCRIPTION
**Pull request by tbursztyka**

### CODE REVIEW

**Comment by SebastianBoe** [comment]
Should this reference to RAW also be removed?

```
/home/sebo/zephyr/boards/arm/usb_kw24d512/Kconfig.defconfig
109:if IEEE802154_MCR20A || IEEE802154_MCR20A_RAW
132:endif # IEEE802154_MCR20A || IEEE802154_MCR20A_RAW
```

**Comment by SebastianBoe** [comment]
Removing myself as reviewer because I still haven't figured out how Kconfig should be written.

**Comment by tbursztyka** [comment]
/boards/arm/usb_kw24d512/Kconfig.defconfig  indeed

**Comment by carlescufi** [comment]
This PR is required for the upcoming OpenThread one.

File: drivers/ieee802154/Kconfig [hunk 1]
```diff
@@ -9,11 +9,15 @@
 #
 # IEEE 802.15.4 options
 #
-menu "IEEE 802.15.4 drivers options"
+menuconfig IEEE802154
+	bool "IEEE 802.15.4 drivers options"
+	default n
+	default y if NET_L2_IEEE802154
```
Comment (SebastianBoe) [inline drivers/ieee802154/Kconfig]: Funny, just did the same change ...

I'll merge this into the OpenThread branch ...

**Comment by nashif** [comment]
recheck

**Review by carlescufi — APPROVED** [review]

**Comment by carlescufi** [comment]
@SebastianBoe are you OK with this? looks good to me

**Comment by SebastianBoe** [comment]
LGTM

**Pull request merged**